### PR TITLE
modified tests

### PR DESCRIPTION
**Pull request by OscarSnowPlow**
## What type of PR is this? (check all applicable)

- [ ] 🍕 Feature
- [x] 🐛 Bug Fix
- [ ] 📝 Documentation Update
- [ ] 🎨 Style
- [ ] 🧑‍💻 Code Refactor
- [ ] 🔥 Performance Improvements
- [x] ✅ Test
- [ ] 🤖 Build
- [ ] 🔁 CI
- [ ] 📦 Chore (Release)
- [ ] ⏩ Revert

## Related Tickets & Documents

https://snplow.atlassian.net/browse/PE-6608

It was highlighted that if you set the parameters up in a particular way (described in ticket above) the code would run successfully as intended but if you were to run a dbt test 8 column tests would fail because those columns are not showing with this particular set up. 

To fix I modified the columns test which were all not null test from the yaml files. 
I replaced these with condition which only runs the test if a particular specification is met.

## Checklist

- [ ] 💣 Is your change a breaking change?
- [ ] 📖 I have updated the CHANGELOG.md

### Added tests?

- [x] 👍 yes
- [ ] 🙅 no, because they aren't needed
- [ ] 🙋 no, because I need help

### Added to documentation?

- [ ] 📓 internal package docs (ymls, macros, readme, if applicable)
- [ ] 📕 I have raised a [Snowplow documentation](https://github.com/snowplow/documentation) PR if applicable (Link here)
- [ ] 🙅 no documentation needed

##  Are there any post-deployment tasks we need to perform?
Maybe an update on documentation.

### CODE REVIEW

**Comment by OscarSnowPlow** [comment]
I have tested on my side feel free to checkout the branch and run your own test 

Let me know if you have any questions or suggestions

**Review by agnessnowplow — COMMENTED** [review]
Thanks for the refactor, looking good! I had a look at the ticket and went through my comments on part 2 for which I didn't see any changes here. I realised that we use derived_tstamp to calculate the absolute_time_in_sec which should be readily available regardless of what they have tracked/enabled. In the end it is essentially the sessions_this_run that is the correct one where we do not tie this column to a variable, the test limitation should also be removed. It is the views_this_run where we should really take it out of the if enabled xyz [clause](https://github.com/snowplow/dbt-snowplow-unified/blob/e09fc4b1c94429408110a41ca61e635662809a53/models/views/scratch/snowplow_unified_views_this_run.sql#L144) to make the behaviour equal to that of sessions. Then the test limitation on that should also be removed. I don't think this is a breaking change, if anything this would be a new field in very rare scenarios.

File: models/sessions/scratch/sessions_scratch.yml [hunk 1]
```diff
@@ -41,7 +41,8 @@ models:
       - name: engaged_time_in_s
         description: '{{ doc("col_session_engaged_time_in_s") }}'
         tests:
-          - not_null
+          - not_null:
```
Comment (agnessnowplow) [inline models/sessions/scratch/sessions_scratch.yml:44]: Let's change the base to the release branch!

Comment (OscarSnowPlow) [reply models/sessions/scratch/sessions_scratch.yml:44]: ah ok will do 
Is this the release branch?
Release/snowplow-unified/0.5.0

Comment (agnessnowplow) [reply models/sessions/scratch/sessions_scratch.yml:44]: yes, I think we can release this together with the Spark updates.

**Pull request merged**